src/main/java/io/bf2/kafka/authorizer/CustomAclAuthorizer.java

```diff
@@ -422,22 +423,17 @@ public List<CompletionStage<AclCreateResult>> createAcls(AuthorizableRequestCont
                 final CompletionStage<AclCreateResult> result;
 
                 if (!binding.entry().principal().startsWith(CustomAclBinding.USER_TYPE_PREFIX)) {
-                    /* Reject ACL operations as invalid where the principal named in the ACL binding is the principal performing the operation */
                     log.info("Rejected attempt by user {} to create ACL binding with invalid principal name: {}",
                             requestContext.principal().getName(),
                             binding.entry().principal());
-                    result = errorResult(AclCreateResult::new, CREATE_ACL_INVALID_PRINCIPAL);
-                } else if (Objects.equals(toString(requestContext.principal()), binding.entry().principal())) {
-                    /* Reject ACL operations as invalid where the principal named in the ACL binding is the principal performing the operation */
-                    log.info("Rejected attempt by user {} to self-assign ACL binding",
-                            requestContext.principal().getName());
-                    result = errorResult(AclCreateResult::new, CREATE_ACL_INVALID_PRINCIPAL);
-                } else if (hasPrincipalBindings(binding.entry().principal())) {
+                    result = errorResult(AclCreateResult::new, CREATE_ACL_INVALID_NON_USER_PRINCIPAL);
+                }
+                else if (hasPrincipalBindings(binding.entry().principal())) {
                     /* Reject ACL operations as invalid where the principal named in the ACL binding is a principal with configured custom ACLs */
                     log.info("Rejected attempt by user {} to create ACL binding for principal {} with existing custom ACL configuration",
                             requestContext.principal().getName(),
                             binding.entry().principal());
-                    result = errorResult(AclCreateResult::new, CREATE_ACL_INVALID_PRINCIPAL);
+                    result = errorResult(AclCreateResult::new, CREATE_ACL_INVALID_PRINCIPAL_MANAGED);
```

Comment:
```suggestion
                    result = errorResult(AclCreateResult::new, INVALID_ACL_PRINCIPAL_MANAGED);
```
?

Reply:
renamed them to `INVALID_ACL_PRINCIPAL_RESTRICTED` and `INVALID_ACL_PRINCIPAL_NON_USER_PREFIXED`